apps/aecore/src/aec_db.erl

```diff
@@ -411,10 +411,11 @@ transactions_by_account(AcctPubKey) ->
     ?t([T || #aec_tx{tx = T}
                  <- mnesia:index_read(aec_tx, AcctPubKey, {acct2tx})]).
 
--dialyzer({nowarn_function, transactions_by_account/2}). %% For mnesia patches.
-transactions_by_account(AcctPubKey, Filter) ->
-    ?t([T || #aec_tx{tx = T}
-        <- mnesia:index_read(aec_tx, AcctPubKey, {acct2tx}), Filter(T)]).
+-dialyzer({nowarn_function, transactions_by_account/3}). %% For mnesia patches.
+transactions_by_account(AcctPubKey, Filter, ShowPending) ->
+    ?t([T || #aec_tx{tx = T, key={_, Where}}
+        <- mnesia:index_read(aec_tx, AcctPubKey, {acct2tx}), Filter(T) 
+              andalso not (ShowPending =:= false andalso Where =:= [])]).
```

Comment:
I see the semantics of where is explicit here https://github.com/aeternity/epoch/blob/4e96464f8be73205919e68e237433b0e3df88c8c/apps/aecore/src/aec_db.erl#L386-L389
Ok.